pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -50,7 +50,7 @@ type DataQueryJson struct {
 }
 
 type DataSource struct {
-	Settings   *models.CloudWatchSettings
+	Settings   models.CloudWatchSettings
```

Comment:
Why not use a pointer here? There's no guarantee this is a lightweight struct with just primitives. Also, the DataSourceInstanceSettings in the PluginContext is a pointer, so think it makes sense that this is a pointer too.

Reply:
I think we only need to load the settings once and we never really need to pass it around?